crates/swc/src/plugin.rs

```diff
@@ -44,12 +44,14 @@ pub struct PluginContext {
 #[cfg(feature = "plugin")]
 pub fn plugins(
     resolver: Option<CachingResolver<NodeModulesResolver>>,
+    comments: Option<&'_ swc_common::comments::SingleThreadedComments>,
```

Comment:
I think we can use `Option<swc_common::comments::SingleThreadedComments>` instead, as all `Comments` impls are cheap to clone.

Reply:
I thought we wanted to keep reference without clone all time. Using cloned would make things lot easier without new scoped tls to hold references :lolsob:

Reply:
By the way, if we clone comments how can we make mutable work for the comments?

Reply:
We use `Rc<RefCell<T>>`.

https://github.com/swc-project/swc/blob/24ce355d77e083957225721a3f3c6fd77ccc8abb/crates/swc_common/src/comments.rs#L349-L357

Reply:
oh right, forgot that completely, which was main reason I couldn't attach into hostenvironment.

Reply:
You can create a wrapper type for comments and do `unsafe impl Send for WrappedComments` + `unsafe impl Sync for WrappedComments` because we are sure that it's alive, and it's not multi threaded.

Reply:
That was the way we created `PseudoScopedKey` for the `GLOBALS` in the plugin context. It is only able to be imported in plugin context (with target_arch support). In this case I was bit hesitant, as `SingleThreadComment` is being used elsewhere than the plugin's context. Even though we think it's likely not happen, any import site can attempt to use SingleThreadedComment with Send + Sync if we create wrapped type for it. Choose explicit approach in this PR instead.